### DISCUSSION TITLE
Is it possible to test downloading file

### DISCUSSION THREAD

**Discussion by finlaydotb** [Orange-OpenSource/hurl#2078] [Q&A] [answered]
As the title says, it is possible to test an endpoint that has content disposition as attachment, ie an endpoint that prompts a download of a file?

**Comment by jcamiel** [comment, marked as answer]
Hi @finlaydotb 

Can you give us more details about waht you want to test?

In any case, if you have an endpoint which is returning the following:

```
200 OK
Content-Type: text/html; charset=utf-8
Content-Disposition: attachment; filename="cool.html"
Content-Length: 21

<HTML>Save me!</HTML>
```

With an Hurl file, you can test that the header `Content-Disposition` has the right value. You can also extract the filename and make sur that you can download the file:


```hurl
# Test that our page return the right HTTP header and
# capture the file name
GET https://foo.com/download.html
HTTP 200
[Asserts]
header "Content-Disposition" contains "cool.html"
[Captures]
file: header "Content-Disposition" regex /filename="(.*)"/

# Do the download!
GET https://foo.com/{{file}}
HTTP 200
Content-Length: 23456 # somple implicit test on headers
```


You can also simulate uploading files with [`[MultipartFormData]` section](https://hurl.dev/docs/request.html#multipart-form-data)
